### PR TITLE
Add resource to POST response

### PR DESCRIPTION
**Pull request by PhilipGlover**
Response body for initial POST (for all 3 PSC types) shows for example:
```
{
    "id": "6411da1fc442594a20e744d3",
    "links": {
        "self": "/transactions/185559-401316-788915/persons-with-significant-control/individual/6411da1fc442594a20e744d3",
        "validation_status": "/transactions/185559-401316-788915/persons-with-significant-control/6411da1fc442594a20e744d3/validation_status"
    },
    "register_entry_date": "2023-02-09",
    "ceased_on": "2023-02-09",
    "reference_etag": "8cb9d4c9ba82059ff01ed638105bc41c1bc4db40",
    "reference_psc_id": "1kdaTltWeaP1EB70SSD9SLmiK5Y"
}
```

- Http status code 201 unchanged
- Location Header unchanged and contains self link

PSC-147

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/pscfiling/api/controller/PscIndividualFilingController.java [hunk 1]
```diff
@@ -28,7 +29,7 @@ public interface PscIndividualFilingController {
      * @throws NotImplementedException implementing classes must perform work
      */
     @PostMapping
-    default ResponseEntity<Object> createFiling(@PathVariable("transactionId") final String transId,
+    default ResponseEntity<PscCommunal> createFiling(@PathVariable("transactionId") final String transId,
```
Comment (hepsimo) [inline src/main/java/uk/gov/companieshouse/pscfiling/api/controller/PscIndividualFilingController.java:32]: Is it possible to return `ResponseEntity<PscIndividualFiling>` instead?

File: src/main/java/uk/gov/companieshouse/pscfiling/api/controller/PscWithIdentificationFilingControllerImpl.java [hunk 2]
```diff
@@ -56,7 +57,7 @@ public PscWithIdentificationFilingControllerImpl(final TransactionService transa
      */
     @Override
     @PostMapping(produces = {"application/json"}, consumes = {"application/json"})
-    public ResponseEntity<Object> createFiling(@PathVariable("transactionId") final String transId,
+    public ResponseEntity<PscCommunal> createFiling(@PathVariable("transactionId") final String transId,
```
Comment (hepsimo) [inline src/main/java/uk/gov/companieshouse/pscfiling/api/controller/PscWithIdentificationFilingControllerImpl.java:60]: Is it possible to return `ResponseEntity<PscWithIdentificationFiling>` instead?

Comment (PhilipGlover) [reply src/main/java/uk/gov/companieshouse/pscfiling/api/controller/PscWithIdentificationFilingControllerImpl.java]: amended and re run code ok

File: src/main/java/uk/gov/companieshouse/pscfiling/api/controller/PscIndividualFilingController.java (hunk 1 above)
Comment (PhilipGlover) [reply src/main/java/uk/gov/companieshouse/pscfiling/api/controller/PscIndividualFilingController.java]: amended and re run code ok

**Review by hepsimo — APPROVED** [review]
Changes look good

**Pull request merged**